examples/metrics-http/src/main/java/io/opentelemetry/example/httpserver/instrumented/InstrumentedHttpServer.java

```diff
@@ -0,0 +1,104 @@
+package io.oopentelemetry.example.httpserver.instrumented;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import io.opentelemetry.api.metrics.GlobalMetricsProvider;
+import io.opentelemetry.api.metrics.LongValueRecorder;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.api.metrics.common.LabelsBuilder;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+/**
+ * An instrumented version of an HttpServer.
+ *
+ * <p>This class demonstrates how someone could instrument the sun http library with metrics that
+ * are later coonsumed by a client. Theoretically this would be an API (like spring-web) that users
+ * are using to write HTTP servers. This code is expected to be written by library authors, not by
+ * Application developers.
+ *
+ * <p>For example, if using spring-web, this code woudl already been written by Spring framework and
+ * not live in the user's application code at all. We pretend this Handler class is part of the
+ * public API of some framework.
+ */
+public class InstrumentedHttpServer {
+  private HttpServer actualServer;
+  /** The meter to use in our fake HTTP server to grab all metrics types. */
+  private static final Meter httpMeter =
+      // Pull metrics from the "global" OpenTelemetry.  This is because we are
+      // a "library" author, and we aren't configuring how/where metrics are exported.
+      GlobalMetricsProvider
+          // Construct a meter that lets users know the metrics came from our library, and
+          // not their own code.  This allows downstream filtering/altering of "built-in" metrics
+          // a bit easier, as you can find/filter what you need from various sources, similar to
+          // "Logger.getLogger(name)" in logging libraries.
+          .getMeter("io.opentelemetry.example.metrics-http.server", "0.1.0");
+
+  /** The description of the metric where we'll be storing durations values. */
+  private static final LongValueRecorder durationMetric =
+      httpMeter
+          .longValueRecorderBuilder("http.server.duration")
+          .setDescription("Length of time for handling an HTTP request")
+          .setUnit("ms")
+          .build();
+
+  public InstrumentedHttpServer(int port) throws IOException {
```

Comment:
why not inject the `MeterProvider` API here, so the user has the option to bring their own, rather than force the use of the global?

Reply:
Great question.  What do you think is the more likely adoption path (or what will we recommend of lirbaries).

Imagine this is "netty" and not "Josh's copy-pasta of the trace http example", what would we expect them to do for their users?

Reply:
I think if people are building library instrumentation (vs. auto-instrumentation via bytecode injection), they should definitely accept the various *Providers to be injected, and only fall back to using the global. This should definitely be our recommendation (and I think we've written that down somewhere as well).

Reply:
Yeah - and technically they would generally just accept an `OpenTelemetry` - once metrics is stable that will have a `MeterProvider` as well.

Reply:
So I guess library instrumentation will generally be

```
newBuilder()
  .setOpenTelemetry(otel)
  // If alpha-level library, exposes alpha level class in public API though this will go away
  .setMeterProvider(meterProvider)
  .build();
```